fuel-merkle/src/binary/verify.rs

```diff
@@ -9,32 +9,34 @@ use crate::{
     },
 };
 
-fn path_length_from_key(key: u64, num_leaves: u64) -> usize {
-    let mut path_length = 0;
-    while (1u64 << path_length) < num_leaves {
-        path_length += 1;
-
-        if path_length >= 64 {
-            break;
-        }
+/// Returns None if:
+/// - `num_leaves` is 0
+/// - the result wouldn't fit into an usize
+fn path_length_from_key(key: u64, num_leaves: u64) -> Option<usize> {
```

Comment:
Because the only place where this is used, it's compared to `proof_set.len()`. I don't understand what you mean by 

> The proof set may contain fewer leaves than `num_leaves` and still be valid and fit into the memory.